pkg/recutil/reconcile.go

```diff
@@ -93,6 +98,19 @@ func ResolveAndReconcile(ctx context.Context, logger kitlog.Logger, mgr manager.
 		logger = logging.WithRecorder(logger, mgr.GetRecorder("theatre"), obj)
 		logger.Log("event", EventStart, "msg", "Starting reconciliation")
 
+		// If the object is being deleted then don't attempt any further
+		// reconciliation, as this can lead to recreating child resources (which
+		// we'd expect to be eventually deleted via propagation) and getting stuck
+		// in an infinite loop, due to these resources now blocking the deletion of
+		// the parent.
+		// If we need to support finalizers in the future then this will need to be
+		// extended to call a function that performs the finalizer actions.
+		if !obj.GetDeletionTimestamp().IsZero() {
+			logger.Log("event", EventSkipped, "msg", "Skipping reconciliation due to deletion")
+			res = reconcile.Result{Requeue: false}
+			return res, nil
+		}
```

Comment:
This will interfere with the console job deletion PR and we will never here that the job has been deleted...

I think we should just add predicates on everything else that ignores the delete event. It also makes it explicit and not happen under the hood in a place you wouldn't expect it?

Reply:
I don't think that it does right?
`ResolveAndReconcile` is only called in the context of the Console object in the console controller. We don't reconcile Jobs directly, we just watch jobs for changes and enqueue a reconciliation for the owning console when there's changes.

Reply:
Discussed in-person and agreed this is fine